webapp/src/containers/WalletPage/components/Tokens/TokensList/index.tsx

```diff
@@ -58,38 +79,108 @@ const WalletTokensList: React.FunctionComponent<WalletTokensListProps> = (
+
   function paginate(
     pageNumber,
     tokensList?: IToken[],
     appTokens: IToken[] = []
   ) {
-    let clone = cloneDeep(tokensList || accountTokens);
+    let clone = cloneDeep(
+      tokensList || [...accountTokens, ...walletTableData, ...verifiedTokens]
+    );
     const keys = {};
     clone.forEach((t) => {
       keys[t.hash] = t.symbol;
     });
     appTokens = (appTokens || []).filter((t) => !keys[t.hash]);
     clone = [...clone, ...appTokens]
       .sort((a: IToken, b: IToken) => +a.hash - +b.hash)
-      .filter((t: IToken) => t.hash != DFI_SYMBOL);
-    const tableData = clone.slice(
+      .filter((t: IToken) => t.hash != '0');
+    const allTokenArray = clone.map((token) => token.symbolKey);
+    if (allTokenArray.length !== [...new Set(allTokenArray)].length) {
+      const duplicateArray = allTokenArray.filter(
+        (value, index) => allTokenArray.indexOf(value) === index
+      );
+      updateWalletToken(duplicateArray);
+    }
+    const tableData = [...new Set(clone)].slice(
       (pageNumber - 1) * pageSize,
       pageNumber * pageSize
     );
     setCurrentPage(pageNumber);
     settableData(tableData);
   }
 
-  useEffect(() => {
-    const verifiedTokens = cloneDeep<IToken[]>(tokens || []).filter((t) => {
-      t.amount = 0;
-      return t.isDAT && !t.isLPS;
+  const handleDropDown = (
+    hash: string,
+    field1: string,
+    symbol: string,
+    field2: string,
+    name: string
+  ) => {
+    const tokenInfo = {
+      symbol: symbol,
+      symbolKey: symbol,
+      amount: '0.00000000',
+      hash: hash,
+      address: '',
+      name: name,
+    };
+    const walletTokensList = handleAddToken(tokenInfo);
+    setwalletTableData(walletTokensList);
+    setFormState({
+      ...formState,
+      [field1]: hash,
+      [field2]: symbol,
     });
-    const tokensList: IToken[] = filterByValue(accountTokens, '');
-    paginate(defaultPage, tokensList, verifiedTokens);
-  }, [accountTokens, isLoadingTokens, props.walletBalance]);
+  };
+
+  const getTokenForWalletDropDown = () => {
+    const existingTokenArray = [
+      ...accountTokens,
+      ...walletTableData,
+      ...verifiedTokens,
+    ].map((value) => value.symbolKey);
+    const filteredTokenMap = new Map<string, any>();
+    tokenData.forEach((value, key) => {
+      if (!existingTokenArray.includes(key)) {
+        filteredTokenMap.set(key, value);
+      }
+    });
+    return filteredTokenMap;
+  };
+
+  const isWalletDropdown = () => {
+    const existingTokenArray = [
+      ...accountTokens,
+      ...walletTableData,
+      ...verifiedTokens,
+    ].map((value) => value.symbolKey);
+    let tokenDataArray: any[] = [];
+    tokenData.forEach((value, key) => {
+      tokenDataArray = [...tokenDataArray, value.symbolKey];
+    });
+    const diffreneceArray = difference(existingTokenArray, tokenData);
+    return diffreneceArray.length > 1 ? true : false;
```

Comment:
No need for true/false here. diffreneceArray.length > 1 would return boolean result. Minor typo on diffreneceArray  -> difference